### ISSUE TITLE
KeyError while building

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#515]
I launched: make html and I received the following output:

```
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.0.3
loading pickled environment... done
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100&#37;] functions                                             
Exception occurred:
  File "/usr/local/lib/python2.6/dist-packages/docutils-0.7-py2.6.egg/docutils/nodes.py", line 534, in __getitem__
    return self.attributes[key]
KeyError: 'module'
The full traceback has been saved in /tmp/sphinx-err-kR_vCt.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
Either send bugs to the mailing list at <http://groups.google.com/group/sphinx-dev/>,
or report them in the tracker at <https://bitbucket.org/birkenfeld/sphinx/issues/>. Thanks!
make: *** [html] Errore 1
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/515
- Originally reported by: Michele Lacchia
- Originally created at: 2010-08-27T09:30:51.173

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-08-27 09:36:09+00:00_

I need a bit more information... can you send me the error log mentioned in the error message?

If I had to guess, I would say that a third-party extension you use wasn't updated for 1.0.

**Comment by shimizukawa** [comment]
_From Anonymous on 2010-11-15 23:20:31+00:00_

I'm seeing the same error:

{{{

# Sphinx version: 1.0.4

# Docutils version: 0.7 release

# Jinja2 version: 2.5.5

Traceback (most recent call last):
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/cmdline.py", line 173, in main
    app.build(force_all, filenames)
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/application.py", line 207, in build
    self.builder.build_update()
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/builders/**init**.py", line 198, in build_update
    'out of date' % len(to_build))
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/builders/**init**.py", line 218, in build
    purple, length):
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/builders/**init**.py", line 120, in status_iterator
    for item in iterable:
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/environment.py", line 515, in update_generator
    self.read_doc(docname, app=app)
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/environment.py", line 676, in read_doc
    app.emit('doctree-read', doctree)
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/application.py", line 311, in emit
    results.append(callback(self, *args))
  File "/Users/dsully/li/lib/python2.6/site-packages/sphinx/ext/viewcode.py", line 50, in doctree_read
    modname = signode['module']
  File "/Users/dsully/li/lib/python2.6/site-packages/docutils/nodes.py", line 534, in **getitem**
    return self.attributes[key]
KeyError: 'module'
}}}

**Comment by shimizukawa** [comment]
_From [FrancescAlted](https://bitbucket.org/FrancescAlted) on 2010-12-14 10:32:05+00:00_

Yes.  Same here.  I'm attaching the stderr and a file that works well (reference.rst), but that by applying the next patch, it reproduces the problem:

{{{
#!python

--- reference.rst       2010-12-14 11:26:29.000000000 +0100
+++ reference.rst.nou   2010-12-14 11:25:57.000000000 +0100
@@ -76,7 +76,7 @@
 Utility functions
 =================

-\* **blosc_set_nthreads(nthreads)**
+.. py:function:: **blosc_set_nthreads(nthreads)**
     Set the number of threads that Blosc can use.

```
 Parameters:
```

}}}

Thanks!

**Comment by shimizukawa** [comment]
_From [FrancescAlted](https://bitbucket.org/FrancescAlted) on 2010-12-14 11:14:01+00:00_

After some tries, the problem disappears if I deactivate the 'sphinx.ext.viewcode' extension module in `conf.py`.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-03 18:34:21+00:00_

Should be fixed in <<cset cc8271486b6cabab593778e4e7707c4076c32020>>.  Thanks!
